### PR TITLE
Make it possible to use VS2019 as the compiler in binary jobs

### PR DESCRIPTION
**Pull request by peterjc123**
Tested with https://github.com/pytorch/pytorch/pull/38949 and https://github.com/pytorch/pytorch/pull/38956.

### CODE REVIEW

File: windows/internal/vs2019_install.ps1 [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+$VS_INSTALL_ARGS = @("--nocache","--quiet","--wait", "--add Microsoft.VisualStudio.Workload.VCTools",
+                                                     "--add Microsoft.Component.MSBuild",
+                                                     "--add Microsoft.VisualStudio.Component.Roslyn.Compiler",
+                                                     "--add Microsoft.VisualStudio.Component.VC.CoreBuildTools",
+                                                     "--add Microsoft.VisualStudio.Component.VC.Redist.14.Latest",
+                                                     "--add Microsoft.VisualStudio.Component.VC.Tools.x86.x64")
+
+curl.exe --retry 3 -kL $VS_DOWNLOAD_LINK --output vs_installer.exe
+if ($LASTEXITCODE -ne 0) {
+    echo "Download of the VS 2017 installer failed"
```
Comment (peterjc123) [inline windows/internal/vs2019_install.ps1:11]: nit: 2019

**Comment by peterjc123** [comment]
Please merge after https://github.com/pytorch/pytorch/pull/38957 is landed. cc @soumith @seemethere

File: windows/internal/vs2019_install.ps1 [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+curl.exe --retry 3 -kL $VS_DOWNLOAD_LINK --output vs_installer.exe
+if ($LASTEXITCODE -ne 0) {
+    echo "Download of the VS 2019 installer failed"
+    exit 1
+}
+
+$process = Start-Process "${PWD}\vs_installer.exe" -ArgumentList $VS_INSTALL_ARGS -NoNewWindow -Wait -PassThru
+Remove-Item -Path vs_installer.exe -Force
+$exitCode = $process.ExitCode
+if (($exitCode -ne 0) -and ($exitCode -ne 3010)) {
```
Comment (malfet) [inline windows/internal/vs2019_install.ps1:18]: Can you please copy VSCollect.exe logic here from https://github.com/pytorch/pytorch/blob/master/.circleci/scripts/vs_install.ps1#L25

And this script is not at all used right now, isn't it?

File: windows/internal/vs2019_install.ps1 [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    echo "VS 2017 installer exited with code $exitCode, which should be one of [0, 3010]."
```
Comment (malfet) [inline windows/internal/vs2019_install.ps1:19]:
```suggestion
    echo "VS 2019 installer exited with code $exitCode, which should be one of [0, 3010]."
```

File: windows/internal/vs2019_install.ps1 (hunk 2 above)
Comment (peterjc123) [reply windows/internal/vs2019_install.ps1:18]: Used https://github.com/pytorch/builder/pull/445/files#diff-c51838f8d7c4e93c825b6382dfc0f40dR2 and https://github.com/pytorch/builder/pull/445/files#diff-e168284874ac9b62e0ccaeba8f6b5db1R2.

File: windows/internal/vs2019_install.ps1 (hunk 3 above)
Comment (peterjc123) [reply windows/internal/vs2019_install.ps1:19]: Nice catch

File: windows/internal/vs2019_install.ps1 (hunk 2 above)
Comment (peterjc123) [reply windows/internal/vs2019_install.ps1:18]: > Can you please copy VSCollect.exe logic here from https://github.com/pytorch/pytorch/blob/master/.circleci/scripts/vs_install.ps1#L25

Sound reasonable. Will need to upload them as artifacts in binary jobs,

**Pull request merged**